utility/get_ip_addrs.py

```diff
@@ -0,0 +1,57 @@
+""" Look at Ceph storage pods, return ip-addresses """
+import subprocess
+import shlex
+
+
+def get_ip_addrs_for_ceph_type(pod_type, namespace='openshift-storage'):
+    """
+    Get the ip address for all pods of a given type.
+    Args:
+        pod_type (string): pod type (mon, mgr, osd, all for everbody)
+        namespace (string): namespace in -n field of oc commands
+    Returns:
+        List of ip_addresses for all pods of the type specified
+    """
+    ret_list = []
+    look_for = f"rook-ceph-{pod_type}"
+    ipaddrs = get_ip_addrs(namespace)
+    for ceph_pod in ipaddrs:
+        if ceph_pod[0].startswith(look_for):
+            ret_list.append(ceph_pod[1])
+    return ret_list
+
+
+def get_ip_addrs(namespace='openshift-storage'):
+    """
+    Get the ip addresses of all pods in the namespaces.
+    Args:
+        namespace (string): namespace in -n field of oc commands
+    Returns:
+        List of two item lists of the form [pod_name, ip_address]
+    """
+    ret_value = []
+    oc_cmd = f"oc -n {namespace} get pods -o wide"
+    process = subprocess.Popen(shlex.split(oc_cmd), stdout=subprocess.PIPE)
+    output, error = process.communicate()
+    if error:
+        print(f"Error executing: {oc_cmd}")
+        return ret_value
+    ip_col = -1
+    lines = output.split(b'\n')
+    for entry in lines:
```

Comment:
Also instead of this text processing ,how about just get pod names and use describe pod to get ip. Also you can use custom-columns I guess https://kubernetes.io/docs/reference/kubectl/overview/#custom-columns to get only the field you want. I haven't tried this but just a guess.

Reply:
I will implement this using describe pod.